optuna/terminator/improvement/evaluator.py

```diff
@@ -128,3 +128,49 @@ def _validate_input(
                 "The intersection search space is empty. This condition is not supported by "
                 f"{cls.__name__}."
             )
+
+
+@experimental_class("3.4.0")
+class BestValueStagnationEvaluator(BaseImprovementEvaluator):
+    """Evaluates the stagnation period of the best value in an optimization process.
+
+    This class is initialized with a maximum stagnation period (`max_stagnation_trials`)
+    and is designed toevaluate the remaining trials before reaching this maximum period
+    of allowed stagnation.
+
+    Args:
+        max_stagnation_trials:
+            The maximum number of trials allowed for stagnation.
+    """
+
+    def __init__(
+        self,
+        max_stagnation_trials: int = 30,
+    ) -> None:
+        if max_stagnation_trials < 0:
+            raise ValueError("The number of tolerance of step must not be negative.")
```

Comment:
`The number of tolerance of step` might not remind `max_stagnation_trials`. Could you update the message, for example, as follows:

```suggestion
            raise ValueError("The maximum number of stagnant trials must not be negative.")
```